### PR TITLE
Check for this.object exists, to allow return null, false, 0 and undefined in object

### PR DESCRIPTION
**Pull request by serhiyzablotskiy**

### CODE REVIEW

File: lib/api/actions.js [hunk 1]
```diff
@@ -30,8 +30,8 @@ class Base {
   }
 
   response() {
-    if (this.object) {
-      if (this.object.toResponse) {
+    if (_.has(this, 'object')) {
```
Comment (alexkravets) [inline lib/api/actions.js:33]: Let's just put `this.object = null` in the constructor, to have it always defined, no? These conditions do look not so good? What are the cases there?

**Review by alexkravets — CHANGES_REQUESTED** [review]

File: lib/api/actions.js (hunk 1 above)
Comment (serhiyzablotskiy) [reply lib/api/actions.js:33]: @alexkravets  We have the endpoint which returns number or null in the object. So if object is 0 or null this condition `if (this.object)` doesn't pass and the value which is returned by this endpoint is an empty array (`_.map(undefined)` returns empty array) which is unexpected behavior.
I have made these changes to have the possibility to return null, 0, false and undefined in the object if required.

**Comment by alexkravets** [comment]
Please provide endpoints that do return number or null, it’s not a valid response, object should always be an object.

- A

**Comment by alexkravets** [comment]
Sergey, what endpoints are you talking about? We should not return number, null or undefined as an object.

**Pull request closed without merging**